File: java/src/test/java/com/iteso/nintendo/characters/SquirtleTest.java
package com.iteso.nintendo.characters;

import com.iteso.nintendo.characters.Squirtle;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.assertEquals;

/**
 * Created by Isaac Cabrera on 2/24/18.
 */
public class SquirtleTest {

    Squirtle squirtle;
    
    @Before
    public void setUp() {
        squirtle = new Squirtle();
    }

    /** Main attack test. */
    @Test
    public void maintAttackTest() {
        assertEquals("Attacking opponent with Small Fire causing a damage of 7", squirtle.mainAttack());
    }

    /** Second attack test. */
    @Test
    public void secondAttackTest() {
        assertEquals("Attacking opponent with Big Fire causing a damage of 13", squirtle.secondAttack());
    }

    /** Defense test */
    @Test
    public void defendTest() {
        assertEquals("Defending attack, damage caused is 0 new HP is 77", squirtle.defend(0));
    }

    /** Name test */
    @Test
    public void getNameTest() {
        assertEquals("Squirtle", squirtle.getName());
    }

    /** Evolve test */
    @Test
    public void evolveTest() {
        assertEquals(null, squirtle.evolve());
    }

    /** Type test */
    @Test
    public void typeTest() {
        assertEquals("agua", squirtle.getType());
    }

    /** Evolvution test */
    @Test
    public void hasEvolutionTest() {
        assertEquals(true, squirtle.hasEvolution());
    }

    /** Main Damage test */
    @Test
    public void getMainAttackDamageTest() {
        assertEquals(7, squirtle.getMainAttackDamage());
    }

    /** Second Damage test */
    @Test
    public void getSecondAttackDamageTest() {
        assertEquals(13, squirtle.getSecondAttackDamage());
    }

    /** Hit points test */
    @Test
    public void getHitPointsTest() {
        assertEquals(77, squirtle.getHitPoints());
    }

    /** Defense multiplier test */
    @Test
    public void getDefenseMultiplierTest() {
        assertEquals(0.3, squirtle.getDefenseMultiplier(), 0.0);
    }
}

File: python/src/unittest/python/characters/Squirtle_test.py
from mockito import mock, verify, when
import unittest
from characters.Squirtle import Squirtle


class Squirtle_test(unnitest.TestCase):
    def setUp(self):
        self.character = Squirtle()

    def test_mainAttack(self):
        self.assertEqual(
            "Attacking opponent with Small Fire causing a damage of 7",
            self.character.funMainAttack())

    def test_secondAttack(self):
        self.assertEqual(
            "Attacking opponent with Big Fire causing a damage of 13",
            self.character.funSecondAttack())

    def test_defend(self):
        self.assertEqual(
            "Defending attack, damage caused is 1 new HP is 76",
            self.character.defend(5))

    def test_getName(self):
        self.assertEqual("Bulbasur", self.character.getName())

    def test_evolve(self):
        self.assertEqual(None, self.character.evolve())

    def test_type(self):
        self.assertEqual("planta", self.character.getType())

    def test_hasEvolution(self):
        self.assertEqual(False, self.character.isHasEvolution())

    def test_getMainAttackDamage(self):
        self.assertEqual(7, self.character.getMainAttackDamage())

    def test_getSecondAttackDamage(self):
        self.assertEqual(13, self.character.getSecondAttackDamage())

    def test_getHitPoints(self):
        self.assertEqual(77, self.character.getHitPoints())

    def test_getDefenseMultiplier(self):
        self.assertEqual(0.3, self.character.getDefenseMultiplier(), 0.0)
